core/services/orm/tree_traversal/ModelObjNode.php

```diff
@@ -127,6 +127,28 @@ public function toArray()
         }
         return $tree;
     }
+
+    /**
+     * @since $VID:$
+     * @return array|mixed
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws ReflectionException
+     * @throws \EE_Error
+     */
+    public function getIds()
+    {
+        $ids = [
+            $this->model_obj->get_model()->get_this_model_name() => [
+                $this->model_obj->ID() => $this->model_obj->ID()
```

Comment:
Will using the ID for the value cause issues if the `relation_nodes` array does not have a corresponding key for this model (which would result in this value being left as is)?

Would it be better to maybe use an empty array for the value because then any consuming code does not have to handle different types? 
ie: consuming code merely has to check if the array is empty or not

Reply:
I'm not sure if I totally grok but here goes...
The values are basically unused; what matters is that the top-most key is a model name; and then the 2nd level keys are the model object IDs.
I don't think consuming code needs to handle different types for the values because (so far) the 2nd-level values are always IDs.